src/GeneralStateTestsFiller/stEWASMTests/README.md

```diff
@@ -0,0 +1,317 @@
+## eWasm Tests
+
+Current [eWasm test cases][1] are based on the main [Ethereum test cases][2]
+which are used by all clients. This allows us to run eWasm tests using
+cpp-ethereum/testeth.
+
+In the main directory of the ewasm tests repository, we can see different kind
+of tests, for example: ABITests, BasicTests, BlockchainTests, GeneralStateTests,
+VMTests, etc. For testing eWasm we are using
+GeneralStateTests. The [GeneralStateTests][3] contains subdirectories grouping
+together tests cases specific for different scenarios.
+
+## Test files
+
+In [GeneralStateTests][3] you can find the test cases specific for
+eWasm: [stEWASMTests][4], this directory contains the "filled" json files, those
+files are read by testeth, testeth reads the [pre-state][5],
+the [transaction][6] and executes it and at the and compares if the postState
+hash is the same as the [expected][7]. These "filled" json files are generated
+by testeth and are based on the [filler files][8] (Note that the filler files
+are inside the [src][15] directory)
+
+## Fillers
+
+Filler files can be written in json or in yaml format, all tests specific for
+eWasm are written in yaml.
+
+The yaml file name should be the same as the test name + `Filler`
+([example][9]), this is validated by testeth.
+
+The test contains 4 main elements:
+
+### 1. env - environmental parameters
+
+For almost all the ewasm tests the environmental parameters remains the same,
+unless we need to test something specific like getting the block gas limit,
+block number, etc.
+
+env example:
+
+```yaml
+env:
+    currentCoinbase: 2adc25665018aa1fe0e6bc666dac8fc2697ff9ba
+    currentDifficulty: '0x020000'
+    currentGasLimit: '89128960'
+    currentNumber: '1'
+    currentTimestamp: '1000'
+    previousHash: 5e20a0453cecd065ea59c37ac63e079ee08998b6045136a8ce6635c7912ec0b6
+```
+
+### 2. pre - state before the transaction (pre-state)
+
+Here we define accounts and the initial values for each one, for almos all test
+cases we are using the account `a94f5374fce5edbc8e2a8697c15331677e6ebf0b` this
+is the account which starts the initial transaction, its [secret key][10] is used in
+the transaction definition. Then we can add more accounts that can be called by
+the transaction.
+
+Each account contains he following fields:
+
+- balance
+- code (empty for Externally Owned Accounts)
+- nonce
+- storage
+
+#### code
+
+Example of wast code: 
+
+```lisp
+(module
+    (import "ethereum" "getCallDataSize" (func  $getCallDataSize (result i32)))
+    (import "ethereum" "storageStore" (func $storageStore (param i32 i32)))
+    (memory 1)
+    (export "memory" (memory 0))
+    (export "main" (func $main))
+    (func $main
+      (i32.store (i32.const 0) (call $getCallDataSize))
+      (call $storageStore (i32.const 100) (i32.const 0))
+    )
+  )
+```
+
+Code is wrapped in a `module`
```

Comment:
This probably should say that "whenever testeth detects a source starting with `(module` as wasm and will automatically compile it"